syncpack.config.cjs

```diff
@@ -0,0 +1,81 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+module.exports = {
+	indent: "\t",
+	workspace: false,
+	// semverGroup rules are applied in order to package/dep combinations. First matching rule applies. When running
+	// `syncpack lint-semver-ranges`, the output is grouped into numbered groups. However, the numbers of the groups are
+	// the _inverse_ of their order in this array. That is, Semver Group 1 in the output will correspond to the _last_
+	// rule in this array.
+	semverGroups: [
+		{
+			// PropertyDDS packages' dependencies are ignored because they use a lot of exact deps.
+			dependencies: ["**"],
+			packages: ["@fluid-experimental/property-*"],
+			isIgnored: true,
+		},
+		{
+			// These dependencies should always be on exact versions
+			dependencies: [
+				"@tiny-calc/*",
+				"@graphql-codegen/cli",
+				"@graphql-codegen/typescript",
+				"@types/chrome",
+				"@types/codemirror",
+				"@types/expect-puppeteer",
+				"@types/jest-environment-puppeteer",
+				"@types/jest",
+				"@types/puppeteer",
+				"@types/url-parse",
+				"fake-indexeddb",
+				"json-stringify-safe",
+				"tinylicious",
+				"yargs",
+			],
+			packages: ["**"],
+			range: "",
+		},
+		{
+			// Some dependencies, like typescript and eslint, recommend to use tilde deps because minors introduce
+			// changes that may break linting
+			dependencies: [
+				"eslint-plugin-*",
+				"eslint-config-prettier",
+				"eslint",
+				"less",
+				"prettier",
+				"typescript",
+				"vue",
+				"webpack-dev-server",
+			],
+			packages: ["**"],
+			range: "~",
+		},
+		{
+			// All deps should use caret ranges unless previously overridden
+			dependencies: ["**"],
+			packages: ["**"],
+			range: "^",
+		},
+	],
+	versionGroups: [
+		{
+			// Some dependencies, like typescript and eslint, recommend to use tilde deps because minors introduce
+			// changes that may break linting
+			dependencies: [
+				"eslint-plugin-*",
+				"eslint-config-prettier",
+				"eslint",
+				"prettier",
+				"typescript",
+				"vue",
+				"webpack-dev-server",
+			],
+			packages: ["**"],
+			range: "~",
```

Comment:
FYI this line won't do anything since this entry is under versionGroups and not semverGroups, you might actually be able to remove this versionGroup and the behaviour will be the same as far as I can tell. Shout if I can help with anything confusing about syncpack 👍

Reply:
Hi @JamieMason, thanks for stopping by! 👋🏻  And thanks for writing/maintaining a great infrastructure tool!

> this line won't do anything since this entry is under versionGroups and not semverGroups

Thank you, I missed that! I removed this and also started using the new custom types feature. Very useful! I may have found a bug which I'll point out in a separate comment.

> Shout if I can help with anything confusing about syncpack

Is my understanding of the groups terminal output correct? Namely that they're numbered, but that the numeric order is the inverse of the array order in the config? When building the syncpack config, I found it very confusing to map the output to the group/rule (I think of them as "rules", like linting). Because a given package/dep can only fit into one group, it was difficult to debug why a given combo winds up in a group I didn't expect.

When you don't know what the state of your repo is -- what packages have mismatched deps that you want to ignore, etc. -- then it helps tremendously if it's easy to map syncpack's output -- _this_ package has _this_ dep that is mismatched -- to the "rule" that is being applied. A feature that may help is to add an optional identifier/name field to the semverGroups and versionGroups, and use those names in the terminal output/logs.

One minor note: we use an unusual semver range for our in-workspace dependencies. For example, `">=2.0.0-internal.3.0.0 <2.0.0-internal.3.1.0"`. Unless I ignore these dependencies with an ignored versionGroup, syncpack overrwrites the versions with `""`. Maybe a bug, but could also be expected/undefined behavior since the version range is unusual.

Reply:
> Is my understanding of the groups terminal output correct? Namely that they're numbered, but that the numeric order is the inverse of the array order in the config? When building the syncpack config, I found it very confusing to map the output to the group/rule (I think of them as "rules", like linting). Because a given package/dep can only fit into one group, it was difficult to debug why a given combo winds up in a group I didn't expect.
> 
> When you don't know what the state of your repo is -- what packages have mismatched deps that you want to ignore, etc. -- then it helps tremendously if it's easy to map syncpack's output -- this package has this dep that is mismatched -- to the "rule" that is being applied. 

Yeah 100% agree, it's weird how it is at the moment – I've created https://github.com/JamieMason/syncpack/issues/120 to update that.

> A feature that may help is to add an optional identifier/name field to the semverGroups and versionGroups, and use those names in the terminal output/logs.

Good idea, I've created https://github.com/JamieMason/syncpack/issues/118 for that.

> One minor note: we use an unusual semver range for our in-workspace dependencies. For example, ">=2.0.0-internal.3.0.0 <2.0.0-internal.3.1.0". Unless I ignore these dependencies with an ignored versionGroup, syncpack overrwrites the versions with "". Maybe a bug, but could also be expected/undefined behavior since the version range is unusual.

That sounds like a bug, I've created https://github.com/JamieMason/syncpack/issues/119.

Thanks a lot for this feedback @tylerbutler, lots of good stuff has come out of it.